stratum/docs/target_options.md

```diff
@@ -0,0 +1,85 @@
+# TargetOptions Class
+
+## Background
+
+Three changes were made to common files in order to support the TDI targets.
+
+1. When built for the DPDK target, `ConfigMonitoringService::DoSet()` no
+   longer pushes the updated chassis configuration to the switch.
+
+2. When built for the DPDK target, `P4Service::SetForwardingPipelineConfig()`
+   returns an error if an attempt is made to overwrite an existing pipeline.
+
+3. `ConfigMonitoringService::PushSavedChassisConfig()` returns an error if
+   the configuration file path is a symlink. This change was made to mitigate
+   a security weakness.
+
+Changes 1 and 2 are controlled by `#ifdef DPDK_TARGET`. Change 3 is
+not conditionalized. It is currently disabled via `#if 0` because it breaks
+unit test cases that use symlinks to manage the test environment.
+
+## Problem statement
+
+- The Stratum code has very few compile-time conditionals, none of which are
+  in common code. The DPDK_TARGET conditionals go against this convention.
+
+- Compile-time conditionals complicate the Bazel build and lead to
+  recompilation when we change change targets. This is undesirable,
```

Comment:
we change targets?

Right now it's change change

Reply:
Oops. Fixed.